src/matching/linear-programming.jl

```diff
@@ -9,6 +33,10 @@ inmatch: `inmatch[e]=true` if edge `e` belongs to the matching.
 m:       `m[i]=j` if vertex `i` is matched to vertex `j`.
          `m[i]=-1` for unmatched vertices.
 """
+maximum_weight_maximal_matching{T<:Number}(g::Graph, w::Dict{Edge,T}, cutoff) = error("JuMP is required")
```

Comment:
I don't think this is right. If you want to error out, then you shouldn't use the same function signature, because then if you have JuMP you'll get a warning about an overwritten function. I think you need

```
maximum_weight_maximal_matching(a...) = error ...
```
instead. (I could be wrong here.)